### PR TITLE
[update:execute] run trailing post updates if there is no updates

### PR DESCRIPTION
**Pull request by lalop**
We can have trailing postUpdates to run or entities update without having any module update.
This pr try to fix that

### CODE REVIEW

**Comment by lalop** [comment]
do you think this can be merged ?

File: composer.json [hunk 1]
```diff
@@ -41,7 +41,7 @@
         "composer/installers": "~1.0",
         "doctrine/annotations": "^1.2",
         "doctrine/collections": "^1.3",
-        "drupal/console-core": "1.8.0",
```
Comment (LOBsTerr) [inline composer.json]: Can you revert this changes ? The version should be fixed

File: src/Command/Update/ExecuteCommand.php [hunk 2]
```diff
@@ -151,42 +154,49 @@ protected function execute(InputInterface $input, OutputInterface $output)
...
+                return 1;
+            }
         }
+        
 
+        // Post Updates are only safe to run after all schemas have been updated.
         if (!$this->getUpdates()) {
+            $postUpdates = $this->postUpdateRegistry->getPendingUpdateInformation();
+            
+            if(!empty($postUpdates)) {
```
Comment (LOBsTerr) [inline src/Command/Update/ExecuteCommand.php]: Space is missing

File: src/Command/Update/ExecuteCommand.php [hunk 3]
```diff
@@ -151,42 +154,49 @@ protected function execute(InputInterface $input, OutputInterface $output)
...
+                watchdog_exception('update', $e);
+                $this->getIo()->error($e->getMessage());
+                return 1;
+            }
         }
+        
 
+        // Post Updates are only safe to run after all schemas have been updated.
         if (!$this->getUpdates()) {
+            $postUpdates = $this->postUpdateRegistry->getPendingUpdateInformation();
```
Comment (LOBsTerr) [inline src/Command/Update/ExecuteCommand.php]: It looks like you set maintenance mode, but it will be reverted only when where are $postUpdates. In other case the website will stay in maintenance mode.
Actually, there is better mechanism for  maintenance mode in the Console core, it is merged, but it is not tagged yet. I asked to create a new tag.

File: src/Command/Update/ExecuteCommand.php [hunk 4]
```diff
@@ -271,9 +281,8 @@ private function runUpdates(
     /**
      * @return bool
      */
-    private function runPostUpdates()
+    private function runPostUpdates($postUpdates)
```
Comment (LOBsTerr) [inline src/Command/Update/ExecuteCommand.php]: Based on the comment above, we can keep it as it was

File: src/Command/Update/ExecuteCommand.php (hunk 3 above)
Comment (LOBsTerr) [reply src/Command/Update/ExecuteCommand.php]: Check the way we will activate maintenance mode

```
protected function configure()
    {
        $this
            ->setName('update:entities')
            ->setDescription($this->trans('commands.update.entities.description'))
            ->enableMaintenance()
            ->setAliases(['upe']);
    }

```

File: composer.json (hunk 1 above)
Comment (lalop) [reply composer.json]: done

File: src/Command/Update/ExecuteCommand.php (hunk 2 above)
Comment (lalop) [reply src/Command/Update/ExecuteCommand.php]: fixed

File: src/Command/Update/ExecuteCommand.php (hunk 3 above)
Comment (lalop) [reply src/Command/Update/ExecuteCommand.php]: it looks much better with this new helper.
I had updated the code, do you want me to rebase the pr ?

**Comment by LOBsTerr** [comment]
@lalop Thank you for your work, no need to rebase. I will merge as soon as we set a new tag on the Drupal console core

**Comment by lalop** [comment]
@LOBsTerr do you have any idea of when does drupal console core will be tagged ?

**Comment by LOBsTerr** [comment]
I have merged it will be tagged soon.
